cmd/ci-operator/main.go

```diff
@@ -864,6 +866,50 @@ func inputHash(inputs api.InputDefinition) string {
 	return oneWayNameEncoding.EncodeToString(hash.Sum(nil)[:5])
 }
 
+// saveNamespaceArtifacts is a best effort attempt to save ci-operator namespace artifacts to disk
+// for review later.
+func (o *options) saveNamespaceArtifacts() {
+	if len(o.artifactDir) == 0 {
+		return
+	}
+
+	namespaceDir := filepath.Join(o.artifactDir, "build-resources")
+	if err := os.Mkdir(namespaceDir, 0777); err != nil {
+		log.Printf("Unable to create build-resources directory: %v", err)
+		return
+	}
+
+	if kubeClient, err := coreclientset.NewForConfig(o.clusterConfig); err == nil {
+		pods, _ := kubeClient.Pods(o.namespace).List(meta.ListOptions{})
+		data, _ := json.MarshalIndent(pods, "", "  ")
+		ioutil.WriteFile(filepath.Join(namespaceDir, "pods.json"), data, 0644)
+		events, _ := kubeClient.Events(o.namespace).List(meta.ListOptions{})
+		data, _ = json.MarshalIndent(events, "", "  ")
+		ioutil.WriteFile(filepath.Join(namespaceDir, "events.json"), data, 0644)
+	}
+
+	if buildClient, err := buildclientset.NewForConfig(o.clusterConfig); err == nil {
+		builds, _ := buildClient.Builds(o.namespace).List(meta.ListOptions{})
+		data, _ := json.MarshalIndent(builds, "", "  ")
+		ioutil.WriteFile(filepath.Join(namespaceDir, "builds.json"), data, 0644)
+	}
+
+	if imageClient, err := imageclientset.NewForConfig(o.clusterConfig); err == nil {
+		if err != nil {
+			return
+		}
+		imagestreams, _ := imageClient.ImageStreams(o.namespace).List(meta.ListOptions{})
+		data, _ := json.MarshalIndent(imagestreams, "", "  ")
+		ioutil.WriteFile(filepath.Join(namespaceDir, "imagestreams.json"), data, 0644)
+	}
+
+	if templateClient, err := templateclientset.NewForConfig(o.clusterConfig); err == nil {
+		templateInstances, _ := templateClient.TemplateInstances(o.namespace).List(meta.ListOptions{})
+		data, _ := json.MarshalIndent(templateInstances, "", "  ")
+		ioutil.WriteFile(filepath.Join(namespaceDir, "templateinstances.json"), data, 0644)
```

Comment:
This can't leak any secrets, can it?

Reply:
Only if we put secrets in there, which we don't today.